judge/models/profile.py

```diff
@@ -124,6 +126,12 @@ class Profile(models.Model):
                                       help_text=_('32 character base32-encoded key for TOTP'),
                                       validators=[RegexValidator('^$|^[A-Z2-7]{32}$',
                                                                  _('TOTP key must be empty or base32'))])
+    scratch_codes = EncryptedNullCharField(max_length=255, null=True, blank=True, verbose_name=_('scratch codes'),
+                                           help_text=_('JSON array of 16 character base32-encoded codes \
+                                                        for scratch codes'),
+                                           validators=[RegexValidator(r'^["\[\],A-Z0-9 ]*$',
```

Comment:
```suggestion
                                           validators=[RegexValidator(r'^(\[\])?$|^\[("[A-Z2-9]{16}", *)*"[A-Z2-9]{16}"\]$',
```
This regex was bothering me, so I wrote a new one that should perfectly match a JSON array.

If we don't care about a perfect JSON match, this regex would work, but it allows trailing commas (which would `json.loads` to error) :
```
^$|^\[("[A-Z2-9]{16}"(, *)?)*\]$
```